### ISSUE TITLE
Midnight Bug

### ISSUE THREAD

**Issue by samvawa** [jsGanttImproved/jsgantt-improved#186]
**Describe the bug**
Tasks ending at midnight end at the next midnight in the javascript version.

**To Reproduce**

Add a task that ends at midnight to the table.

I saw the issue entering dates using both `YYYY-MM-dd HH:mm:00` and `new Date(YYYY-MM-dd HH:mm:00)` formats

**Expected behavior**
Have it end at midnight on the correct day.

**Screenshots**

![image](https://user-images.githubusercontent.com/37078531/56148534-809d0880-5f78-11e9-9c46-2bd1afe783c0.png)

Note that the start/end date is correct but the bar's end extends the extra day.

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows 10
 - Browser [e.g. chrome, safari] Firefox
 - Version [e.g. 22] 66

**Additional context**

Commenting out
```
if ((curTaskEnd.getTime()-(curTaskEnd.getTimezoneOffset()*60000))%(86400000)==0) curTaskEnd=new Date(curTaskEnd.getFullYear(), curTaskEnd.getMonth(), curTaskEnd.getDate()+1, curTaskEnd.getHours(), curTaskEnd.getMinutes(), curTaskEnd.getSeconds()); // add 1 day here to simplify calculations below
```
in jsgantt.js under `// Draw each row` fixed the issue.

**Comment by mariohmol** [comment]
thanks!

one thing that is strange in this chart.. month is not ending on the last day,.. stops on 23 of april..  and have day with 0...


can u do a PR? this change fixes all?

**Comment by samvawa** [comment]
The screenshot is with an hourly display; 23 is hour 23 on April 27. The issue was in the day and week view as well. 

I saw that if-check and `curTaskEnd.getDate()+1` in one other place, though it seems like it wasn't causing the same issue.

I also wasn't sure if I should make a PR since it has the comment `// add 1 day here to simplify calculations below` - I didn't want to make the update without having the more complex calculations "below" if they were necessary.

**Comment by mariohmol** [comment]
fixed on 2.1.19!

Thanks
